### PR TITLE
[#2064] Extract AbstractSubscription out of CommandSubscription

### PR DESCRIPTION
**Pull request by calohmn**
For #2064:
This is in preparation of adding support for different kinds of MQTT subscriptions, notably a subscription on an error topic.

### CODE REVIEW

File: adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/CommandSubscriptionsManager.java [hunk 1]
```diff
@@ -70,18 +70,25 @@ public CommandSubscriptionsManager(final Vertx vertx, final T config) {
      * Invoked when a device sends an MQTT <em>PUBACK</em> packet.
      *
      * @param msgId The message/packet id of the command published with QoS 1.
+     * @return {@code true} if a pending command request for the PUBACK was found.
      * @throws NullPointerException if msgId is {@code null}.
      */
-    public void handlePubAck(final Integer msgId) {
+    public boolean handlePubAck(final Integer msgId) {
```
Comment (kaniyan) [inline adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/CommandSubscriptionsManager.java:76]: I see that the return value is not being used anywhere. Is it here in consideration with the upcoming PRs?

File: adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/CommandSubscriptionsManager.java [hunk 2]
```diff
@@ -70,18 +70,25 @@ public CommandSubscriptionsManager(final Vertx vertx, final T config) {
...
+                .map(pendingCommandRequest -> {
+                    LOG.trace("acknowledgement received for command sent to device [packet-id: {}]", msgId);
                     if (pendingCommandRequest.timerId != null) {
                         cancelTimer(pendingCommandRequest.timerId);
                     }
                     pendingCommandRequest.onAckHandler.handle(msgId);
-                }, () -> LOG.debug("no active command request found for received acknowledgement [packet-id: {}]", msgId));
+                    return true;
+                }).orElseGet(() -> {
+                    if (!subscriptions.isEmpty()) {
```
Comment (kaniyan) [inline adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/CommandSubscriptionsManager.java:87]: Is this check needed? It seems to me that this log message makes sense even when there are no subscriptions (i.e the _map_ is empty) .

File: adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/CommandSubscriptionsManager.java (hunk 1 above)
Comment (calohmn) [reply adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/CommandSubscriptionsManager.java:76]: Yes, right. The PUBACK could have been sent for a published command *or* a published message on the new error topic for #2064. So, the PUBACK isn't necessarily only handled in the CommandSubscriptionsManager any more.

File: adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/CommandSubscriptionsManager.java (hunk 2 above)
Comment (calohmn) [reply adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/CommandSubscriptionsManager.java:87]: This is an extra check in preparation for the upcoming error topic handling (see also other comment).
A PUBACK received while `subscriptions` is empty is most certainly not a PUBACK for a command message (but instead for a published error message for example).

**Review by kaniyan — APPROVED** [review]
LGTM

**Pull request merged**